tools/clientSimulator/pfClientSimulator.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/python3
+# Copyright (c) 2014-2015, Intel Corporation
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+# list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation and/or
+# other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from criterion.CriterionClassFactory import CriterionClassFactory
+from testGenerator.TestVectorFactory import TestVectorFactory
+from testGenerator.TestLauncher import TestLauncher
+from testGenerator.SubprocessLogger import SubprocessLoggerThread
+from testGenerator.SubprocessLogger import ScriptLoggerThread
+from configuration.ConfigParser import ConfigParser
+from scenario.Scenario import Scenario
+from userInteraction.UserInteractor import UserInteractor
+from userInteraction.DynamicCallHelper import DynamicCallHelper
+import argparse
+import threading
+import signal
+import time
+import logging
+import os
+
+
+def close(logger, testLauncher, coverage):
```

Comment:
Please split in two functions: one that handles "graceful" termination, calling the another one that forces termination.

This will also make it possible to ensure that the second one is called - e.g. at the bottom of this source file, after main() returns.